### PR TITLE
fix: skip bot comment when user skips comments tour

### LINKED ISSUE #7281: [Bug]Skip bot comments if the comments tour is skipped
## Description
Skip bot comments if the comments tour is skipped

### Steps to reproduce the behaviour:

1. Signup with a new user
2. Click on toggle comment mode button
3. Click on the skip button, enter the name and submit the comments onboarding form
4. Add a new comment, a bot comment is added, if the tour is skipped the bot comment should also be skipped

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
Removed the client dependency on locally stored state. As of now the client changes have been added, the BE changes are still in progress.

Fixes #7281 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- manual
- automated tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/skip-bot-comments 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.82 **(0.01)** | 36.95 **(0)** | 33.81 **(0.01)** | 55.44 **(0.02)**
 :red_circle: | app/client/src/actions/userActions.ts | 67.86 **(-0.06)** | 100 **(0)** | 0 **(0)** | 83.33 **(-3.03)**
 :red_circle: | app/client/src/api/UserApi.tsx | 64.91 **(0.1)** | 0 **(0)** | 5.56 **(-0.32)** | 64.29 **(0.14)**
 :red_circle: | app/client/src/pages/Editor/ToggleModeButton.tsx | 80.99 **(0.99)** | 51.39 **(-3.02)** | 80 **(0)** | 80.15 **(0.45)**
 :red_circle: | app/client/src/reducers/uiReducers/usersReducer.ts | 20.59 **(-0.62)** | 0 **(0)** | 0 **(0)** | 20.59 **(-0.62)**
 :green_circle: | app/client/src/sagas/userSagas.tsx | 16.52 **(0.23)** | 1.45 **(0)** | 5.26 **(-0.3)** | 19.79 **(0.22)**
 :green_circle: | app/client/src/utils/storage.ts | 34.48 **(0.73)** | 16.67 **(0)** | 10 **(0.91)** | 26.85 **(0.8)**</details>

### CODE REVIEW

File: app/client/src/constants/userConstants.ts [hunk 1]
```diff
@@ -2,13 +2,19 @@ export const ANONYMOUS_USERNAME = "anonymousUser";
 
 type Gender = "MALE" | "FEMALE";
 
+export enum CommentsOnBoardingState {
```
Comment (sbalaji1192) [inline app/client/src/constants/userConstants.ts:5]: We should name it as `CommentsOnboardingState` with small b in onboarding

**Review by sbalaji1192 — APPROVED** [review]

**Review by sbalaji1192 — APPROVED** [review]
LGTM

**Comment by pranavkanade** [comment]
BE changes - https://github.com/appsmithorg/appsmith/pull/7816

**Review by rishabhsaxena — APPROVED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**